ui/src/components/ColorPicker/CustomColorPicker.js

```diff
@@ -0,0 +1,191 @@
+    marginBottom: '12px',
+  },
+  Hue: {
+    radius: '2px',
+  },
+  palette: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+  },
+  picker: {
+    width: '225px',
+    background: '#fff',
+    borderRadius: '2px',
+    boxShadow: '0 0 2px rgba(0,0,0,.3), 0 4px 8px rgba(0,0,0,.3)',
+    boxSizing: 'initial',
+  },
+  saturation: {
+    width: '100%',
+    paddingBottom: '55%',
+    position: 'relative',
+    borderRadius: '2px 2px 0 0',
+    overflow: 'hidden',
+  },
+  Saturation: {
+    radius: '2px 2px 0 0',
+  },
+  swatch: {
+    width: '20px',
+    height: '20px',
+    borderRadius: '0px',
+    marginBottom: '12px',
+  },
+});
+
+/* eslint new-cap: ["error", { "capIsNew": false }] */
+const CustomColorPickerInner = ColorWrap((props) => {
+  const styles = buildStyles(props);
+  return (
+    <div style={styles.picker}>
+      <div style={styles.saturation}>
+        <Saturation
+          style={styles.Saturation}
+          hsl={props.hsl}
+          hsv={props.hsv}
+          pointer={ChromePointerCircle}
+          onChange={props.onChange} />
+      </div>
+
+      <div style={styles.body}>
+        <div style={styles.hue}>
+          <Hue
+            style={styles.Hue}
+            hsl={props.hsl}
+            pointer={ChromePointer}
+            onChange={props.onChange} />
+        </div>
+
+        <div style={styles.controls}>
+          <div style={styles.color} />
+
+          <div style={styles.hex} >
+            <EditableInput
+              style={{ wrap: styles.HEXwrap, input: styles.HEXinput, label: styles.HEXlabel }}
+              label="hex"
+              value={props.hex}
+              onChange={props.onChange} />
+          </div>
+        </div>
+
+        <div style={styles.palette} >
+          {
+            props.customColors.map(c => (
+              <Swatch
+                key={c}
+                color={c}
+                hex={c}
+                onClick={props.onChange}
+                style={styles.swatch}
+                focusStyle={{
+                  boxShadow: `0 0 4px ${c}`,
+                }} />
+            )).valueSeq()
+          }
+        </div>
+
+        <div style={styles.checkMark} >
+          <button
+            className="btn btn-primary btn-sm"
+            onClick={props.onClickCheckMark}>
+            <i className="ion ion-checkmark" />
+          </button>
+        </div>
+      </div>
+    </div>
+  );
+});
+
+class CustomColorPicker extends React.PureComponent {
+
+  static propTypes = {
+    // immutable.List<tinycolor.ColorInput>
+    // https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/tinycolor2/index.d.ts
+    customColors: React.PropTypes.instanceOf(List),
+
+    // (color: tinycolor.Instance) => void
+    onClickCheckMark: React.PropTypes.func
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      color: colorHelper.toState(props.customColors.first('#000000'))
```

Comment:
Default value of immutable.Collection.first() was added on v4.0.0-rc.10. The version `^3.8.1`, we use, does not have it.
https://github.com/immutable-js/immutable-js/releases/tag/v4.0.0-rc.10